name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/wizards/datatransfer/CSVImportDefinitionPage.java

```diff
@@ -515,14 +521,35 @@ public Color getBackground(Object element, int columnIndex)
                 if (column.getFormat() != null)
                 {
                     String text = getColumnText(element, columnIndex);
-                    if (text != null)
+                    if (text != null && !text.isEmpty())
+                    {
                         column.getFormat().getFormat().parseObject(text);
+                        return resources.createColor(GREEN);
+                    }
+                    else
+                    {
+                        if (column.getField().isOptional())
+                        {
+                            return resources.createColor(LIGHTGREEN);
+                        }
+                        else
+                        {
+                            return resources.createColor(ORANGE);
+                        }
+                    }
+                }
+                else
+                {
+                    String text = getColumnText(element, columnIndex);
+                    return resources.createColor(GREEN);
```

Comment:
Felder, die kein Format habe, können aber trotzdem Optional oder nicht optional sein. Aktuell wird das aber nur für Felder mit Format angezeigt (z.B. ISIN), aber nicht für die WKN oder das Ticker Symbol, was ja eigentlich identisch ist. Ich schlage vor hier auch die optionale Farbmarkierung anzuwenden.

Reply:
Abend Andreas,
ok, habe das in 9b02ee2 implementiert.
Grüße,
cmaoling